### PR TITLE
Build scala-cli with jvm 11

### PR DESCRIPTION
**Pull request by tpasternak**
Let's check what happens

### CODE REVIEW

**Review by alexarchambault — COMMENTED** [review]
We need to make sure that the `runner`, `test-runner`, and `stub` modules are still fine with Java 8, by passing them the right javac and scalac options (see [this SO question](https://stackoverflow.com/questions/25926111/how-to-force-sbt-to-use-java-8), I think we need both its javac and scalac options). These are the modules that are sometimes added to the user class path by Scala CLI.

To be sure it works, we can add a simple test in `RunTestDefinitions` say, that passes `--jvm 8` to `scala-cli`.

**Comment by MarcinAman** [comment]
Wouldn't upgrading to latest LTS (JDK 17) make sense? Afaik this is quite a problematic release as oracle decided to encapsulate some of its internals therefore some tooling may not be compatible. So if it works on JDK 17 it will probably run on older JDK-s

**Comment by alexarchambault** [comment]
For now, we're using the Java 11 version of GraalVM to build the native launcher, so switching to Java 11 isn't a problem. Maybe the upcoming GraalVM version will have Java 17 versions, and allow to switch to Java 17 then.

**Comment by tpasternak** [comment]
Graal 17 is out https://phoronix.com/scan.php?page=news_item&px=GraalVM-21.3-Released

File: build.sc [hunk 1]
```diff
@@ -524,12 +525,13 @@ class BloopRifle(val crossScalaVersion: String) extends CrossSbtModule with Scal
...
 class TastyLib(val crossScalaVersion: String) extends CrossSbtModule with ScalaCliPublishModule
     with ScalaCliScalafixModule {
-  def scalacOptions = T {
+  def scalacOptions = T(
     super.scalacOptions() ++ {
       if (scalaVersion().startsWith("2.")) Seq("-Ywarn-unused")
       else Nil
-    }
-  }
+    } ++ Seq("-release", "8")
```
Comment (alexarchambault) [inline build.sc]: I think that should have been added to the `stubs` module. This one doesn't need it.

File: modules/integration/src/test/scala/scala/cli/integration/NativePackagerTests.scala [hunk 2]
```diff
@@ -214,7 +214,7 @@ class NativePackagerTests extends munit.FunSuite {
       // format: off
       val cmd = Seq[os.Shellable](
         TestUtil.cli,
-        "package", helloWorldFileName,
+        "package", "--jvm", "8", helloWorldFileName,
```
Comment (alexarchambault) [inline modules/integration/src/test/scala/scala/cli/integration/NativePackagerTests.scala]: Maybe we should just change the base image in [`Package.scala`](https://github.com/VirtusLab/scala-cli/blob/a95f8925f636cc8bb4efadeae9c0c7942c85465f/modules/cli/src/main/scala/scala/cli/commands/Package.scala#L355).

File: modules/integration/src/test/scala/scala/cli/integration/TestTestDefinitions.scala [hunk 3]
```diff
@@ -509,5 +509,4 @@ abstract class TestTestDefinitions(val scalaVersionOpt: Option[String])
         expect(output.contains("Hello from native"))
     }
   }
-
```
Comment (alexarchambault) [inline modules/integration/src/test/scala/scala/cli/integration/TestTestDefinitions.scala:512]: It'd be nice to check that tests run fine with `--jvm 8` too (this should ensure that `test-runner` is fine on Java 8).

File: website/docs/commands/package.md [hunk 4]
```diff
@@ -114,7 +114,7 @@ The docker image name parameter `--docker-image-repository` is mandatory.
 The following command generates a `hello-docker` image with the `latest` tag:
 
 ```bash
-scala-cli package --docker HelloDocker.scala --docker-image-repository hello-docker
+scala-cli package --jvm 8 --docker HelloDocker.scala --docker-image-repository hello-docker
```
Comment (alexarchambault) [inline website/docs/commands/package.md]: Same as above, this shouldn't be needed with a change of base image.

File: build.sc (hunk 1 above)
Comment (tpasternak) [reply build.sc]: Done

File: modules/integration/src/test/scala/scala/cli/integration/NativePackagerTests.scala (hunk 2 above)
Comment (tpasternak) [reply modules/integration/src/test/scala/scala/cli/integration/NativePackagerTests.scala]: Done, but this does not fully resolve the problem. I opened an issue #413

File: modules/integration/src/test/scala/scala/cli/integration/TestTestDefinitions.scala (hunk 3 above)
Comment (tpasternak) [reply modules/integration/src/test/scala/scala/cli/integration/TestTestDefinitions.scala:512]: Done

File: website/docs/cookbooks/scala-docker.md [hunk 5]
```diff
@@ -18,7 +18,7 @@ Passing `--docker` to the `package` sub-command generates a Docker image. When c
 The following command generates a `hello-docker` image with the `latest` tag:
 
 ```bash
-scala-cli package --docker HelloDocker.scala --docker-image-repository hello-docker
+scala-cli package --jvm 8 --docker HelloDocker.scala --docker-image-repository hello-docker
```
Comment (alexarchambault) [inline website/docs/cookbooks/scala-docker.md]: Do we really need to pass that option here? It should work fine without it (doesn't it?)

File: website/docs/cookbooks/scala-docker.md [hunk 6]
```diff
@@ -51,12 +51,9 @@ Built docker image, run it with
 This command creates a Scala Native Docker image:
 
 ```bash ignore
-scala-cli package --native --docker HelloDocker.scala --docker-image-repository hello-docker
+scala-cli package --native --jvm 8 --docker HelloDocker.scala --docker-image-repository hello-docker
```
Comment (alexarchambault) [inline website/docs/cookbooks/scala-docker.md]: Same as above.

**Review by alexarchambault — APPROVED** [review]

**Pull request merged**